routers/repo/issue.go

```diff
@@ -465,16 +465,26 @@ func ViewIssue(ctx *context.Context) {
 	}
 	ctx.Data["Title"] = fmt.Sprintf("#%d - %s", issue.Index, issue.Title)
 
-	iw, exists, err := models.GetIssueWatch(ctx.User.ID, issue.ID)
-	if err != nil {
-		ctx.Handle(500, "GetIssueWatch", err)
-		return
-	}
-	if !exists {
+	var iw *models.IssueWatch
+	var exists bool
+	if ctx.User != nil {
+		iw, exists, err = models.GetIssueWatch(ctx.User.ID, issue.ID)
+		if err != nil {
+			ctx.Handle(500, "GetIssueWatch", err)
+			return
+		}
+		if !exists {
+			iw = &models.IssueWatch{
+				UserID:     ctx.User.ID,
+				IssueID:    issue.ID,
+				IsWatching: models.IsWatching(ctx.User.ID, ctx.Repo.Repository.ID),
+			}
+		}
+	} else {
 		iw = &models.IssueWatch{
```

Comment:
I prefer that iw would stay at `nil` and in template https://github.com/go-gitea/gitea/blob/master/templates/repo/issue/view_content/sidebar.tmpl#L104 the form is only display if logged or/and IssueWatch is set.